ribbon/src/test/java/feign/ribbon/RibbonClientTest.java

```diff
@@ -172,6 +175,38 @@ public void testFeignOptionsClientConfig() {
     assertEquals(2, config.getProperties().size());
   }
 
+  @Test
+  public void testServiceMatchesUriPart() throws IOException {
+    server1.enqueue(new MockResponse().setBody("success!"));
+
+    String hostAndPort = hostAndPort(server1.getUrl(""));
+    getConfigInstance().setProperty(serverListKey(),
+        hostAndPort);
+
+
+    String baseUrl = "http://" + hostAndPort;
+    getConfigInstance().setProperty(serverListKey(), baseUrl);
+
+    final Holder<String> holder = new Holder<String>();
+    RibbonClient client = RibbonClient.builder()
+        .delegate(new Client() {
+          @Override
+          public Response execute(Request request, Request.Options options) throws IOException {
+            holder.held = request.url();
+            return Response.create(200, "", Collections.EMPTY_MAP, "success", Charset.defaultCharset());
+          }
+        }).build();
+
+    client.execute(new Request("GET", "http://"+client()+"/"+client(), Collections.EMPTY_MAP, new byte[0], Charset.defaultCharset()), new Request.Options(1111, 22222));
```

Comment:
Suggestions on a better way to test this?  One thought was to bring in power mock to mock the final `Request`.  The other would be to refactor the replaceFirst line into a static method that was easy to test.